test/tst-conversion-int32.c

```diff
@@ -6,6 +6,42 @@
 
 #include <smd.h>
 
+#define test(VARTYPE, VARDTYPE) \\
```

Comment:
Nice idea with the macro. However, I still got a comment: Whenever you write a function like macro (a macro that is to be used like `foo(bar);`, that macro should be enclosed in a do-while-loop:

    #define foo(bar) do { \
        int bar_int = (bar); \
        frobnicate(bar_int, bar_int*42); \
    } while(false)

The point of using the `do { ... } while(false)` loop encapsulation is to correctly treat a trailing semicolon in the call. So, for instance when you expand

    if(baz) foo(baz); else abort();

the result is

    if(bar) do { \
        int bar_int = (bar); \
        frobnicate(bar_int, bar_int*42); \
    } while(false); else abort();

Without the `do { ... } while(false)` encapsulation, you get

    if(bar) { \
        int bar_int = (bar); \
        frobnicate(bar_int, bar_int*42); \
    }; else abort();

instead, which is a syntax error because of the extra `;` between the closing brace and the `else`. The do-while-encapsulation swallows the trailing semicolon.

I think, it's best to get into the habit of wrapping multi-statement macros in a do-while-loop immediately, because a) it's a very common idiom in C code, and b) it's the only way to write function like macros that actually behave like function calls syntactically.